packages/trader/src/Stores/Modules/Trading/Helpers/process.js

```diff
@@ -11,7 +11,7 @@ export const processTradeParams = async (store, new_state) => {
     for (let i = 0; i < functions.length; i++) {
         // Shallow copy with Object.assign is good enough to extend the snapshot with new state
         // we don't need deep extension here, since each function in functions array composes a property of the store completely
-        Object.assign(snapshot, await functions[i](snapshot)); // eslint-disable-line no-await-in-loop
+        await Object.assign(snapshot, functions[i](snapshot)); // eslint-disable-line no-await-in-loop
```

Comment:
the functions are async we should await for the call

Reply:
awaiting the object assignment fixed the bug.

Reply:
Well done guys 👍🏼

Reply:
@hirad-rewok it resolves the issue but it may create more issues in future.
Better not to have any await inside a for loop.

Reply:
Moreover Object.assign is a sync call, we don't have to await on it - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

Reply:
We should find a way to make the function in `functions` array synchronous so that we don't have to await on the function calls